pkg/controllers/statusaggregator/plugins/pod.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2023 The KubeAdmiral Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugins
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+
+	"github.com/kubewharf/kubeadmiral/pkg/controllers/common"
+)
+
+type PodPlugin struct{}
+
+func NewPodPlugin() *PodPlugin {
+	return &PodPlugin{}
+}
+
+func (receiver *PodPlugin) AggregateStatuses(
+	ctx context.Context,
+	sourceObject, fedObject *unstructured.Unstructured,
+	clusterObjs map[string]interface{},
+	clusterObjsUpToDate bool,
+) (*unstructured.Unstructured, bool, error) {
+	logger := klog.FromContext(ctx).WithValues("status-aggregator-plugin", "pods")
+
+	needUpdate := false
+	aggregatedStatus := &corev1.PodStatus{}
+	phases := map[corev1.PodPhase][]string{
+		corev1.PodPending:   make([]string, 0, len(clusterObjs)),
+		corev1.PodRunning:   make([]string, 0, len(clusterObjs)),
+		corev1.PodSucceeded: make([]string, 0, len(clusterObjs)),
+		corev1.PodFailed:    make([]string, 0, len(clusterObjs)),
+		// Ignore phase PodUnknown as it has been deprecated since year 2015.
+	}
+
+	for clusterName, clusterObj := range clusterObjs {
+		logger := klog.FromContext(ctx).WithValues("cluster-name", clusterName)
+
+		utd := clusterObj.(*unstructured.Unstructured)
+		// For status of pod
+		var found bool
+		status, found, err := unstructured.NestedMap(utd.Object, common.StatusField)
+		if err != nil || !found {
+			logger.Error(err, "Failed to get status of cluster object")
```

Comment:
We can continue aggregating the statuses of other cluster pods even if status is not found for one of them